android

React-native 0.65.1 React-native MMKV :1.3.1
Failed to build on Android
C/C++: /Users/wangqiang/ttvc/TTvc/node_modules/react-native-mmkv/android/CMakeLists.txt debug|armeabi-v7a : CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
REACT_NATIVE_JNI_LIB
    linked by target "reactnativemmkv" in directory /Users/wangqiang/ttvc/TTvc/node_modules/react-native-mmkv/android

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -29,7 +29,7 @@ if (rootProject.ext.has('reactNativeAndroidRoot')) {
 } else {
   reactNativeDir = new File(
           projectDir,
-          '/../../../node_modules/react-native/android'
+          '/../node_modules/react-native/android'
   )
 }
 if (reactNativeDir.exists()) {

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -128,7 +128,35 @@ dependencies {
   extractJNI(files(rnAAR))
 }
 
-task extractJNIFiles {
+task extractJNIFilesForExternal {
+  doLast {
+    configurations.extractJNI.files.each {
+      def file = it.absoluteFile
+
+      copy {
+        from zipTree(file)
+        into "$buildDir/$file.name"
+        include "jni/**/*"
+      }
+    }
+  }
+}
+
+task extractJNIFilesForDebug {
+  doLast {
+    configurations.extractJNI.files.each {
+      def file = it.absoluteFile
+
+      copy {
+        from zipTree(file)
+        into "$buildDir/$file.name"
+        include "jni/**/*"
+      }
+    }
+  }
+}
+
+task extractJNIFilesForRelease {
   doLast {
     configurations.extractJNI.files.each {
       def file = it.absoluteFile
@@ -144,6 +172,14 @@ task extractJNIFiles {
 
 tasks.whenTaskAdded { task ->
   if (task.name.contains('externalNativeBuild')) {
-    task.dependsOn(extractJNIFiles)
+    task.dependsOn(extractJNIFilesForExternal);
+  }
+
+  if (task.name.contains('generateJsonModelDebug')) {
+    task.dependsOn(extractJNIFilesForDebug);
+  }
+
+  if (task.name.contains('generateJsonModelRelease')) {
+    task.dependsOn(extractJNIFilesForRelease)
   }
 }

